### PR TITLE
[Small] Several bug fixes for MuZero

### PR DESCRIPTION
**Pull request by emailweixu**
Two was introduced by PR 1105: config needs to be passed to subclass init, missing SequentialDataTransformer.members()
One was introduced by PR 1112: prediction_net was created twice

Also changed prepare_rnn_batch_norm() to do warning instead of raising when torch.nn.BatchNorm is found.

### CODE REVIEW

File: alf/batch_norm.py [hunk 1]
```diff
@@ -317,9 +317,9 @@ def prepare_rnn_batch_norm(module: nn.Module) -> bool:
         if isinstance(m, (BatchNorm1d, BatchNorm2d)):
             bns.add(m)
         elif isinstance(m, (nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d)):
-            raise ValueError(
-                "RNN does not support torch.nn.BatchNorm layer "
-                "(at %s). Please use alf.layers.BatchNorm instead." % path)
+            warning_once(
+                "MCTSModel may not perform well with torch.nn.BatchNorm layer "
```
Comment (breakds) [inline alf/batch_norm.py]: I think it might be more general to mention RNN instead of `MCTSModel` when making this recommendation.

Comment (emailweixu) [reply alf/batch_norm.py]: agree.

**Review by breakds — APPROVED** [review]
